### PR TITLE
Ts tips update2

### PR DESCRIPTION
**Pull request by OrchardPerformance**
Update 1 to our tuner studio tool tips

### CODE REVIEW

File: firmware/integration/rusefi_config.txt [hunk 1]
```diff
@@ -418,10 +423,10 @@ end_struct
 custom engine_type_e 4 bits, S32, @OFFSET@, [0:6], @@engine_type_e_auto_enum@@
 engine_type_e engineType;http://rusefi.com/wiki/index.php?title=Manual:Engine_Type\nset engine_type X
 
-int engineSnifferRpmThreshold;Engine sniffer would be disabled above this rpm\nset engineSnifferRpmThreshold X;"RPM", 1, 0, 0, 30000, 0
+int engineSnifferRpmThreshold;+Engine sniffer would be disabled above this rpm\nset engineSnifferRpmThreshold X;"RPM", 1, 0, 0, 30000, 0
 
 struct injector_s
-	float flow;+This is your injector flow at the fuel pressure used in the vehicle. cc/min, cubic centimetre per minute\nBy the way, g/s = 0.125997881 * (lb/hr)\ng/s = 0.125997881 * (cc/min)/10.5\ng/s = 0.0119997981 * cc/min;"cm3/min", 1, 0, 0, 99999, 2
+	float flow;+This is your injector flow at the fuel pressure used in the vehicle. cc/min, cubic centimeter per minute\nBy the way, g/s = 0.125997881 * (lb/hr)\ng/s = 0.125997881 * (cc/min)/10.5\ng/s = 0.0119997981 * cc/min;"cm3/min", 1, 0, 0, 99999, 2
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:429]: not sure we need all the math here, maybe just cc/min to lb/hr conversion?

File: firmware/integration/rusefi_config.txt [hunk 2]
```diff
@@ -482,19 +487,19 @@ MAP_sensor_config_s map;@see isMapAveragingEnabled
...
 	float knockBandCustom;+We calculate knock band based of cylinderBore\n Use this to override - kHz knock band override;"kHz", 1, 0, 0, 20, 2
 
 
-float[DWELL_CURVE_SIZE] sparkDwellRpmBins;On Single Coil or Wasted Spark setups you have to lower dwell at high RPM;"RPM", 1, 0, 0, 18000, 2
+float[DWELL_CURVE_SIZE] sparkDwellRpmBins;+On Single Coil or Wasted Spark setups you have to lower dwell at high RPM;"RPM", 1, 0, 0, 18000, 2
 	float[DWELL_CURVE_SIZE] sparkDwellValues;;"ms", 1, 0, 0, 30, 2
 
 struct_no_prefix specs_s
-float displacement;Engine displacement, in litres\nsee also cylindersCount;"L", 1, 0, 0, 1000, 2
+float displacement;+Engine volume/capacity, in litres\nsee also cylindersCount;"L", 1, 0, 0, 1000, 2
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:500]: what was wrong with `displacement`?

File: firmware/integration/rusefi_config.txt [hunk 3]
```diff
@@ -482,19 +487,19 @@ MAP_sensor_config_s map;@see isMapAveragingEnabled
...
 
-uint32_t cylindersCount;;"", 1, 0, 1, @@MAX_CYLINDER_COUNT@@, 0
+uint32_t cylindersCount;+Number of cylinder the engine has.;"", 1, 0, 1, @@MAX_CYLINDER_COUNT@@, 0
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:502]: `How many cylinders or rotors does your engine have?`

File: firmware/integration/rusefi_config.txt [hunk 4]
```diff
@@ -507,15 +512,15 @@ firing_order_e firingOrder;
...
 int sensorSnifferRpmThreshold;+Disable sensor sniffer above this rpm;"RPM", 1, 0, 0, 30000, 0
-	int rpmHardLimit;set rpm_hard_limit X;"rpm", 1, 0, 0, 20000, 2
+	int rpmHardLimit;+Engine speed to activate hard rpm limit.;"rpm", 1, 0, 0, 20000, 2
 
 #define engine_load_mode_e_enum "INVALID", "INVALID", "INVALID", "Speed Density", "MAF Air Charge", "Alpha-N", "Lua"
 
 
 custom engine_load_mode_e 4 bits, U32, @OFFSET@, [0:2], @@engine_load_mode_e_enum@@
-engine_load_mode_e fuelAlgorithm;+This setting controls which fuel quantity control algorithm is used.\nAlpha-N means drive by TPS\nSpeed Density requires MAP sensor;
+engine_load_mode_e fuelAlgorithm;+This setting controls which fuel quantity control algorithm is used.\nAlpha-N means drive by TPS commonly only used for NA engines\nSpeed Density requires MAP sensor and is the default choice for may installs\nMAF air charge is a cylinder filling based method that uses a mass air flow sensor.;
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:523]: `may` -> typo?

File: firmware/integration/rusefi_config.txt [hunk 5]
```diff
@@ -552,11 +557,11 @@ float vehicleSpeedCoef;+This coefficient translates vehicle speed input frequenc
 custom can_nbc_e 4 bits, U32, @OFFSET@, [0:4], "None", "FIAT", "VAG", "MAZDA RX8", "BMW", "W202", "BMW E90", "Haltech", "VAG MQB", "Nissan VQ35", "Genesis Coupe", "Honda K", "type 12", "type 13", "type 14", "INVALID"
 can_nbc_e canNbcType;set can_mode X
 
-int canSleepPeriodMs;CANbus thread period, ms;"ms", 1, 0, 0, 1000, 2
+int canSleepPeriodMs;+CANbus thread period in ms;"ms", 1, 0, 0, 1000, 2
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:560]: we should double check what this actually does - I think it's just the update rate for the "rusefi verbose CAN" messages

File: firmware/integration/rusefi_config.txt [hunk 6]
```diff
@@ -567,10 +572,10 @@ log_format_e logFormat;
 	int HD44780width;;"index", 1, 0, 0, 300, 0
 	int HD44780height;;"index", 1, 0, 0, 300, 0
 
-	adc_channel_e tps1_1AdcChannel;First throttle body, first sensor. See also pedalPositionAdcChannel
+	adc_channel_e tps1_1AdcChannel;+First throttle body, first sensor. See also pedalPositionAdcChannel
 adc_channel_e vbattAdcChannel;+This is the processor input pin that the battery voltage circuit is connected to, if you are unsure of what pin to use, check the schematic that corresponds to your PCB. 
 adc_channel_e fuelLevelSensor;+This is the processor pin that your fuel level sensor in connected to. This is a non standard input so will need to be user defined. 
-	adc_channel_e tps2_1AdcChannel;Second throttle body position sensor, single channel so far\nset_analog_input_pin tps2 X
+	adc_channel_e tps2_1AdcChannel;+Second throttle body position sensor, single channel so far\nset_analog_input_pin tps2 X
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:578]: `second throttle, primary sensor`
no need for console command here

File: firmware/integration/rusefi_config.txt [hunk 7]
```diff
@@ -638,13 +643,13 @@ end_struct
...
 	brain_pin_e directionPin2;
-	brain_pin_e controlPin;Acts as EN pin in two-wire mode
+	brain_pin_e controlPin;+Acts as EN pin in two-wire mode
 	brain_pin_e disablePin;
 end_struct
 	
 	idle_hardware_s idle;
 
-	float manIdlePosition;value between 0 and 100 used in Manual mode;"%", 1, 0, 0, 100, 0
+	float manIdlePosition;+Value between 0 and 100 used in Manual mode;"%", 1, 0, 0, 100, 0
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:652]: not useful tooltip

File: firmware/integration/rusefi_config.txt [hunk 8]
```diff
@@ -956,11 +961,11 @@ custom maf_sensor_type_e 4 bits, S32, @OFFSET@, [0:1], @@maf_sensor_type_e_enum@
 
 	stft_s stft
 
-	dc_io[DC_PER_STEPPER iterate] stepperDcIo
+	dc_io[DC_PER_STEPPER iterate] stepperDcIol
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:964]: typo, extra `l`

File: firmware/integration/rusefi_config.txt [hunk 9]
```diff
@@ -1131,32 +1136,33 @@ custom pin_mode_e 1 bits, U08, @OFFSET@, [0:6], @@pin_mode_e_enum@@
...
 
 ! todo: mapErrorDetectionIdleTooLow? 30kPa is usually lowest on idle	
-	float mapErrorDetectionTooLow;kPa value which is too low to be true;"kPa", 1, 0, -100, 100, 2
-	float mapErrorDetectionTooHigh;kPa value which is too high to be true;"kPa", 1, 0, -100, 800, 2
+	float mapErrorDetectionTooLow;+kPa value which is too low to be true;"kPa", 1, 0, -100, 100, 2
+	float mapErrorDetectionTooHigh;+kPa value which is too high to be true;"kPa", 1, 0, -100, 800, 2
 	uint16_t multisparkSparkDuration;+How long to wait for the spark to fire before recharging the coil for another spark.;"ms", 0.001, 0, 0, 3, 2
 	uint16_t multisparkDwell;+This sets the dwell time for subsequent sparks. The main spark's dwell is set by the dwell table.;"ms", 0.001, 0, 0, 3, 2
-	pid_s idleRpmPid;See cltIdleRpmBins
+	pid_s idleRpmPid;+See cltIdleRpmBins
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:1152]: tooltips on structs (like this pid_s) don't actually show anywhere, so the `+` does nothing.

File: firmware/integration/rusefi_config.txt [hunk 10]
```diff
@@ -1337,9 +1343,9 @@ custom stepper_num_micro_steps_e 1 bits, U08,	@OFFSET@,	[0:3], @@stepper_num_mic
 	brain_pin_e mc33816_rstb;ResetB
 	brain_pin_e mc33816_driven
 
-	switch_input_pin_e brakePedalPin;Brake pedal switch
-	adc_channel_e cj125ua;lambda input
-	adc_channel_e cj125ur;heater input
+	switch_input_pin_e brakePedalPin;+Brake pedal switch
+	adc_channel_e cj125ua;+lambda input
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:1347]: things like this don't need the `+` - those are truly only useful as code comments, not in-TS comments

**Comment by rusefillc** [comment]
Actually "+" prefix USED to be required until recently

See 
```
    // we used to have a weird feature of only handing comments starting with + as tooltips
    // very unclear what was the value of it. todo; remove all of those and kill this tag?
    public static final char TS_COMMENT_TAG = '+';
```

**Comment by mck1117** [comment]
> 
> 
> Actually "+" prefix USED to be required until recently
> 
> See
> 
> ```
>     // we used to have a weird feature of only handing comments starting with + as tooltips
>     // very unclear what was the value of it. todo; remove all of those and kill this tag?
>     public static final char TS_COMMENT_TAG = '+';
> ```

I thought the point was that you could omit the `+` for it to be an in-code-only comment, or add the + so you see it in TS as well.

**Comment by rusefillc** [comment]
@mck1117 yes absolutely, but that was error prone so i removed. Default should be enabled tooltip, maybe '-' prefix if desired to NOT tooltip. Or just not bother and tooltip always?

**Comment by rusefillc** [comment]
I am not saying that 30 changes should be 30 separate PRs but maybe we are trying to PR too many unrelated things together?

@OrchardPerformance maybe a smaller PR at least for all uncontested changes would be the next good step? Just a suggestion please use your judgement

**Comment by OrchardPerformance** [comment]
Well there is a lot more here to read/check than there was yesterday. 
Works been bonkers.
Will go back through tonight.

File: firmware/integration/rusefi_config.txt (hunk 2 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:500]: Using displacement as a way to explain displacement kind of fails the brief. 
Think looking in the dictionary to see: Displacement - The displacement of an engine.

File: firmware/integration/rusefi_config.txt (hunk 3 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:502]: I guess it does now apply to rotors as well now

File: firmware/integration/rusefi_config.txt (hunk 4 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:523]: yep

File: firmware/integration/rusefi_config.txt (hunk 5 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:560]: It's the comment that was already in there so could be obsolete. 
I was going to activate as many existing tips as possible and then ask everyone to report low quality ones.

File: firmware/integration/rusefi_config.txt (hunk 6 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:578]: set_analog_input_pin tps2 X ?

File: firmware/integration/rusefi_config.txt (hunk 7 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:652]: Do we have a better tip? That one was the existing deactivated tip

File: firmware/integration/rusefi_config.txt [hunk 11]
```diff
@@ -691,28 +696,28 @@ switch_input_pin_e clutchDownPin;+Some cars have a switch to indicate that clutc
...
+	ego_sensor_e afr_type;+AFR, WBO, EGO - whatever you like to call it;
 	float idle_antiwindupFreq;+0.1 is a good default value;"x", 1, 0, -1000000, 1000000, 4
 
 	brain_input_pin_e[TRIGGER_INPUT_PIN_COUNT iterate] triggerInputPins;
 	pin_output_mode_e hip9011CsPinMode;
 	output_pin_e tachOutputPin;
 	pin_output_mode_e tachOutputPinMode;
 
-	output_pin_e mainRelayPin;
+	output_pin_e mainRelayPin;+STM32 pin that the main relay output is connected to
```
[the remark this answers is not in the document]
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:720]: These also apply to frank, do we consider those obsolete now and focus on MRE/Prot/Hellen versions?

File: firmware/integration/rusefi_config.txt (hunk 1 above)
Comment (OrchardPerformance) [reply firmware/integration/rusefi_config.txt:429]: Historic tip so up for review, guess we probably don't but lb/hr conversion would be good, maybe cc/min to g/s as we work in g/sec for some things.

**Comment by OrchardPerformance** [comment]
As far as I can see in TS the only tool tips showing are the ones with a + before them.

File: firmware/integration/rusefi_config.txt (hunk 2 above)
Comment (rusefillc) [reply firmware/integration/rusefi_config.txt:500]: image

@OrchardPerformance please confirm that you see tooltip
@OrchardPerformance please confirm that you do NOT see '+' in master version of this file

**Comment by rusefillc** [comment]
Too much happens within same PR

1) map high/low voltage I am not sure suggestion is right
2) whole drama between + and no +
3) removal of console commands is a separate issues to agree on
4) deciding if we mention stm32 or not
5) lack of branched comments on PR discussion screen
6) just too many things at once, the whole "all or nothing"

I've cherry-picked some of the changes from top of the files until it seemed like too much. That could be reference of how much I can consume in one go.

I am well aware that cherry picking is not great but so is the approach in this PR which I would prefer to close

**Pull request closed without merging**